js/web/lib/wasm/jsep/webgpu/ops/gather.ts

```diff
@@ -105,9 +105,13 @@ export const gather = (context: ComputeContext, attributes: GatherAttributes): v
 
   const metadata = {
     name: 'Gather',
-    inputTypes: [GpuDataType.default, GpuDataType.default],
-    cacheHint: attributes.cacheKey + inputs[0].dataType.toString(10) + inputs[1].dataType.toString(10),
+    inputTypes: [GpuDataType.default],
+    cacheHint: attributes.cacheKey + inputs[0].dataType.toString(10) + inputs[1].dataType.toString(10) +
```

Comment:
input datatype is by default included in the final cache key, no need to add them.

Reply:
fixed